### PR TITLE
Conform to Cran test platform

### PR DESCRIPTION
**Pull request by henrykironde**
In order to pass on cran we changing a few things in the source, mainly how we utilize reticulate,
The are no backward compatibility changes.

### CODE REVIEW

**Comment by henrykironde** [comment]
Checking passes on the windows platform at https://win-builder.r-project.org/
Sample temporary link 
https://win-builder.r-project.org/j9UYohf8034g/00check.log
https://win-builder.r-project.org/MIrqSX9t0FYk/00check.log

**ToDos:**
Clean up some of the code and try to include previous helper functions of checking which version of python or retriever. I Will notify with a `Ready for review` once done

To check locally, one does not need retriever (python installed). 
```
R CMD build .
R CMD check rdataretriever_2.0.0.tar.gz --no-build-vignettes --no-manual --ignore-vignettes --no-examples 
```

**Comment by henrykironde** [comment]
Ready for review

File: README.md [hunk 1]
```diff
@@ -169,6 +181,33 @@ To run tests
 
 `docker-compose  run rdata Rscript load_and_test.R`
 
+Release
+-------
+
+Make sure you have test passing on R-oldrelease, current R-release and R-devel
```
Comment (ethanwhite) [inline README.md]: "have the tests"

File: R/rdataretriever.R [hunk 2]
```diff
@@ -581,3 +513,23 @@ get_os <- function() {
...
+
+# global reference to python modules (will be initialized in .onLoad)
+retriever <- NULL
+
+.onLoad = function(libname, pkgname) {
+    if(reticulate::py_available()){
+        install_python_modules <- function(method = "auto", conda = "auto") {
+            reticulate::py_install("retriever", method = method, conda = conda)
+        }
+    }
```
[the remark this answers is not in the document]
Comment (henrykironde) [reply R/rdataretriever.R]: @ethanwhite thanks for the review and nice remarks. I have to say, getting the package to build on the server was not easy but we were able to build locally(even on a fresh Docker image with `Python` removed). Following the docs did not work well, I had to check the `Reverse imports` of `reticulate` 
Using the described `onLoad` method did not work on the test server https://win-builder.r-project.org/. But thinking about it, may be I could make the `.onLoad` function better and simpler. I will give it a try with [ref to reticulate is_python_initialized method](https://github.com/rstudio/reticulate/blob/439203175dbdc3620474560c8b96cb180dafafa8/R/package.R#L64) .

**Comment by henrykironde** [comment]
> `skip_on_cran()` is included with `skip_if_no_python()`

We skip on `CRAN`: such test will never get tested on `CRAN`.
We do have to add `skip_if_no_python()` for every test function to be able to build on `CRAN` in case test environment on `CRAN` may have no `Python`

some ref: 
skip:  https://github.com/rstudio/reticulate/blob/master/tests/testthat/helper-utils.R#L36
onLoad: https://rstudio.github.io/reticulate/articles/package.html

> 1. In the more basic instructions I wonder if we should replace the `py_install` with `conda_install('r-reticulate', 'retriever'). This would ensure that it ends up in the conda `r-reticulate` environment even if there is another Python install (e.g., on OS X). I feel like I may have run into an issue with `py_install` in a case like this, which is why I'm bringing it up.


Looks like `py_install` calls conda by default.

https://rstudio.github.io/reticulate/reference/py_install.html#details

**Comment by ethanwhite** [comment]
Thanks @henrykironde! This looks good to me and is testing well locally for me as well. In it goes!

**Pull request merged**